src/cdk-experimental/popover-edit/popover-edit-position-strategy-factory.ts

```diff
@@ -24,19 +35,31 @@ export abstract class PopoverEditPositionStrategyFactory {
  */
 @Injectable()
 export class DefaultPopoverEditPositionStrategyFactory extends PopoverEditPositionStrategyFactory {
-  constructor(protected readonly overlay: Overlay) {
+  constructor(
+      protected readonly overlay: Overlay) {
     super();
   }
 
-  forElementRef(elementRef: ElementRef): PositionStrategy {
-    return this.overlay.position().flexibleConnectedTo(elementRef)
+  positionStrategyForCells(cells: HTMLElement[]): PositionStrategy {
+    // We assume that the cells are in DOM order.
+
+    return this.overlay.position()
+        .flexibleConnectedTo(cells[0])
         .withGrowAfterOpen()
         .withPush()
+        .withViewportMargin(10)
         .withPositions([{
           originX: 'start',
           originY: 'top',
           overlayX: 'start',
           overlayY: 'top',
         }]);
   }
+
+  sizeConfigForCells(cells: HTMLElement[]): OverlaySizeConfig {
+    return {
+      width: cells.reduce(
```

Comment:
Couldn't we make this a bit faster by measuring the `ClientRect.left` of the first cell and the `ClientRect.right` of the last one? That way we don't have to keep calling `getBoundingClientRect` in a loop.